modin/core/storage_formats/base/query_compiler.py

```diff
@@ -856,6 +857,142 @@ def merge(self, right, **kwargs):  # noqa: PR02
             self, right=right, **kwargs
         )
 
+    def _get_column_as_pandas_series(self, key):
+        """
+        Get column data by label as pandas.Series.
+
+        Parameters
+        ----------
+        key : Any
+            Column label.
+
+        Returns
+        -------
+        pandas.Series
+        """
+        result = self.getitem_array([key]).to_pandas().squeeze(axis=1)
+        assert isinstance(result, pandas.Series)
```

Comment:
I assume we shouldn't use `assert` in any place except tests.

Reply:
I think that's a controversial position :) In my view, it's better to fail fast when we rely on assumptions that fail. Here we assume we get a series because 1)  `left_by`, `left_on`, `right_by`, `right_on` are all scalars if they're not None 2) they don't represent duplicate column names. 

I think 1) is wrong because we don't check this for `left_on` and `right_on`, which can be lists, and 2) will break with an uninformative `AttributeError: 'DataFrame' object has no attribute 'dtype'` in pandas

But these bugs are out of scope for this PR.

Reply:
Okay, now the code raises `RuntimeError`.